packages/blade/src/components/Drawer/Drawer.web.tsx

```diff
@@ -0,0 +1,206 @@
+    >
+      <FloatingPortal>
+        {isMounted ? (
+          <FloatingFocusManager
+            context={context}
+            initialFocus={initialFocusRef ?? (stackingLevel >= 2 ? backButtonRef : closeButtonRef)}
+          >
+            <Box
+              position="fixed"
+              {...metaAttribute({
+                name: MetaConstants.Drawer,
+                testID,
+              })}
+              // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+              zIndex={zIndex + stackingLevel}
+            >
+              {showOverlay || stackingLevel === 2 ? (
+                <DrawerOverlay
+                  onClick={() => {
+                    onDismiss();
+                  }}
+                  className={isVisible ? SHOW_DRAWER : ''}
+                  lockScroll={true}
+                />
+              ) : null}
+              <AnimatedDrawerContainer
+                className={isVisible ? SHOW_DRAWER : ''}
+                width={{ base: '90%', s: '375px', m: '420px' }}
+                {...makeAccessible({
+                  role: 'dialog',
+                  modal: true,
+                  label: accessibilityLabel,
+                })}
+                position="fixed"
+                top="spacing.0"
+                left="100%"
+                backgroundColor="popup.background.subtle"
+                elevation="highRaised"
+                height="100%"
+                ref={refs.setFloating}
+                onKeyDown={(event) => {
+                  if (event?.key === 'Escape' || event?.code === 'Escape') {
+                    onDismiss();
+                  }
+                }}
+              >
+                {children}
+              </AnimatedDrawerContainer>
+            </Box>
+          </FloatingFocusManager>
+        ) : null}
+      </FloatingPortal>
+    </DrawerContext.Provider>
+  );
+};
+
+/**
+ * ### Drawer Component
+ * 
+ * A drawer is a panel that slides in mostly from right side of the screen over the existing content in the viewport. 
+ * It helps in providing additional details or context and can also be used to promote product features or new products.
+ * 
+ * ---
+ * 
+ * #### Usage
+ * 
+ * ```jsx
+  const MyDrawer = () => {
+    const [showDrawer, setShowDrawer] = React.useState(false);
+    return (
+      <Box>
+        <Button onClick={() => setShowDrawer(true)}>Open Drawer</Button>
+        <Drawer
+          isOpen={showDrawer}
+          onDismiss={() => setShowDrawer(false)}
+        >
+          <DrawerHeader title="Announcements" />
+          <DrawerBody>
+            <FTXAnnouncement />
+            <CatPictures />
+          </DrawerBody>
+        <Drawer>
+      </Box>
+    )
+  }
+ * ```
+ *
+ *  ---
+ *
+ * Checkout {@link https://blade.razorpay.com/?path=/docs/components-drawer Drawer Documentation}
+ * 
+ * 
+ */
+const Drawer = assignWithoutSideEffects(_Drawer, {
+  componentId: drawerComponentIds.Drawer,
```

Comment:
Put this id inside `MetaConstants`

Reply:
MetaConstants is for component attributes no? I have added another entry there. This is for componentIds of individual components

Reply:
We have been using MetaConstants for these too.

Reply:
Not really. In most components I saw we're using the name similar to the displayName of component.

MetaConstants are lower-cased values

<img width="506" alt="image" src="https://github.com/razorpay/blade/assets/30949385/4d5ae24d-b352-4302-b6b2-cc6d9353ffaf">
<img width="729" alt="image" src="https://github.com/razorpay/blade/assets/30949385/3cc86b10-4198-4749-83ee-f5df4851ba5c">
<img width="516" alt="image" src="https://github.com/razorpay/blade/assets/30949385/4fa1967e-8494-402a-a783-9e868e059d63">
<img width="680" alt="image" src="https://github.com/razorpay/blade/assets/30949385/fba8d229-cb8c-4975-8d11-9cd126efe3f0">